keps/sig-api-machinery/2855-automated-stored-versions/README.md

```diff
@@ -0,0 +1,567 @@
+
+Historically, the [kube-storage-version-migrator] has not managed modifying the `status.storedVersion` field because the CRD's storedVersion could change mid-migration unbenownst to the migrator due to slow/failed watches. 
+
+This KEP proposes a solution that would allow the [kube-storage-version-migrator] to safely modify a CRD's `status.storedVersions` after a successful migration.
+
+[kube-storage-version-migrator]: https://github.com/kubernetes-sigs/kube-storage-version-migrator
+
+## Motivation
+
+It is desirable for the [kube-storage-version-migrator] to modify a CRD's `status.storedVersions` array after a successful migration rather than relying on a cluster admin to manually perform the same steps.
+
+### Goals
+
+- Allow the [kube-storage-version-migrator] to update a CRD's `status.storedVersions`  with the list of stored versions after a successful migration.
+- Ensure that the kube-storage-version-migrator has successfully done its job before removing a stored version from the status of the CRD.
+
+### Non-Goals
+
+- Modify a CRD's `status.storedVersions` field if the migration is not successful.
+
+## Proposal
+
+In order to safely modify a CRD's `status.storedVersions` field, the [kube-storage-version-migrator] would need to prevent changes to the CRD during a migration. CRD's could be marked as migrating using an annotation and changes to the CRD could be prevented by means of an [Validating Admission Webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+
+The validating admission webhook would need to:
+
+- Intercept all Update and Patch operations against CRDs that include the `ksvm.k8s.io/migrating=true` annotation.
+- Allow `/status` requests from the [kube-storage-version-migrator] service account.
+- Allow setting or removing the migrating annotation by the [kube-storage-version-migrator] service account.
+- Prevent other changes to the CRD.
+
+A summary of the new migration workflow can be found below:
+
+1. The [kube-storage-version-migrator] notices a new stored version for the `foo` CRD.
+2. The  `ksvm.k8s.io/migrating=true`  annotation is applied to the`foo` CRD that defines the resource being migrated.
+3. The `StorageVersionMigration` CR is created for the resource introduced by the `foo` CRD.
+4. The [kube-storage-version-migrator] begins migrating all `foo` resources as it does today.
+5. When a migration is successful, the `foo` CRD's `status.storedVersion` is patched to only include the new stored version.
+6. The `ksvm.k8s.io/migrating=true` annotation is removed from the `foo` CRD.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a cluster admin, I would like the [kube-storage-version-migrator] to automatically update a CRD's `status.storeVersions` after a successful a migration.
```

Comment:
Waiting to comment until I see a response to [this](https://github.com/kubernetes/enhancements/pull/2856#discussion_r687919657) as it might help define  **when** this happens.